core/templates/services/html-length.service.ts

```diff
@@ -16,85 +16,210 @@
  * @fileoverview Service for Computing length of HTML strings.
  */
 
+export const CALCULATION_TYPE_WORD = 'word';
+export const CALCULATION_TYPE_CHARACTER = 'character';
+
+// eslint-disable-next-line max-len
+const CUSTOM_TAG_REGEX = /<oppia-noninteractive-(?:math|image|link|collapsible|video|skillreview|tabs)[^>]*>/g;
+
+type CalculationType =
+  typeof CALCULATION_TYPE_WORD | typeof CALCULATION_TYPE_CHARACTER;
+
 import { Injectable, SecurityContext } from '@angular/core';
 import { LoggerService } from './contextual/logger.service';
 import { DomSanitizer } from '@angular/platform-browser';
+import { HtmlEscaperService } from './html-escaper.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class HtmlLengthService {
   constructor(
     private loggerService: LoggerService,
-    private sanitizer: DomSanitizer
+    private sanitizer: DomSanitizer,
+    private htmlEscaperService: HtmlEscaperService
   ) {}
 
   /**
     * ['oppia-noninteractive-image', 'oppia-noninteractive-math']
     * The above tags constitutes for non text nodes.
     */
-  nonTextTags = ['oppia-noninteractive-math', 'oppia-noninteractive-image'];
+  nonTextTags = [
```

Comment:
Can you update the above comment to match your code?

Reply:
I will do that. Thanks!